### PR TITLE
WD-2031 Add a no roles message to careers departments with no open roles

### PR DESCRIPTION
**Pull request by anthonydillon**
## Done
- Fixed the logic detecting and showing the no roles message
- Updated the message to match the Vanilla empty state styling
- Run prettier on the JS file

## QA

- Open the demo and go to /careers/all?filter=Administration
- See there is a message warning of no roles
- Click another department and see the message is removed and the roles are listed correctly.

## Screenshot
image

### CODE REVIEW

**Comment by lyubomir-popov** [comment]
@anthonydillon that image is going to be contentious. Can we simply not have an image?

**Comment by anthonydillon** [comment]
We can I just thought it made it nicer

**Comment by anthonydillon** [comment]
Updated to this:
image

File: templates/partial/_careers-available-roles.html [hunk 1]
```diff
@@ -1,7 +1,21 @@
 {% if vacancies %}
-<h2 class="p-heading--4 js-filter__no-results u-hide">
+<!-- <h2 class="p-heading--4 js-filter__no-results u-hide">
```
Comment (mtruj013) [inline templates/partial/_careers-available-roles.html:2]: Left over comment?

**Review by mtruj013 — APPROVED** [review]
LGTM! Just a left over comment that we can delete I think

**Comment by marinacastejon** [comment]
@anthonydillon a few comments:
- I agree with Lyubo's comment, much better without the image
- I think a better and more consistent copy would be: "There are no open roles matching your criteria."
- I see that this also appears in all empty states, could you please update the copy for all cases when there are no results?
Thanks

**Comment by anthonydillon** [comment]
@marinacastejon believe I have addressed your points? Would you mind reviewing it again please?

**Comment by marinacastejon** [comment]
LGTM, thanks @anthonydillon

**Comment by lyubomir-popov** [comment]
@anthonydillon @marinacastejon do we need the filters when there's  nothing to filter? Why not simply do this instead: 
image

**Comment by anthonydillon** [comment]
Applying filters could cause no roles so it would remove the ability to change that.

**Comment by lyubomir-popov** [comment]
@anthonydillon ah I see. Then this? So no p-strip, and just one col offset, then inside - h2 instead of h4: 

image

**Comment by anthonydillon** [comment]
@lyubomir-popov thanks, done.

File: templates/partial/_careers-available-roles.html [hunk 2]
```diff
@@ -1,7 +1,18 @@
...
+  <div class="row">
+    <div class="u-align--left col-6 col-start-large-2">
+      <h2>There are no open roles matching your criteria</h2>
+      <p>Why not try widening your search? You can do this by:</p>
+      <ul class="p-list">
+        <li class="p-list__item is-ticked">Searching for alternative words or phrases</li>
+        <li class="p-list__item is-ticked">Filtering by different department</li>
+        <li class="p-list__item is-ticked">Trying a different spelling</li>
+      </ul>
+      <a href="/careers/all" class="p-button--positive">View all roles</a>
```
Comment (lyubomir-popov) [inline templates/partial/_careers-available-roles.html:12]:
```suggestion
      <a href="/careers/all" class="p-button">View all roles</a>
```
we already have a primary button on this page

**Review by lyubomir-popov — APPROVED** [review]
All good, just one optional suggestion in the code.

**Pull request merged**